app/battle/fight.py

```diff
@@ -0,0 +1,12 @@
+from app.knights.knights import Knight
+
+
+def fight(knight1: Knight, knight2: Knight) -> None:
+    knight1.hp -= knight2.power - knight1.protection
```

Comment:
Ensure that the `protection` attribute is correctly initialized for each knight to prevent negative damage calculations. Consider adding a check to ensure that `knight2.power - knight1.protection` does not result in negative damage.